pkg/image/image.go

```diff
@@ -44,9 +44,16 @@ var registeredFormats = []struct {
 	{"ext3", &ext3Format{}},
 }
 
+// ErrArchMismatch denotes that the image was built for use on a
+// different platform and the host architecture is incompatible.
+var ErrArchMismatch = fmt.Errorf("image arch doesn't match host's")
```

Comment:
Ok it eases detection but we loose information. The previous error message displayed current arch and detected arch in image, we should keep this information and display it as error for users. Why not use the suggestion made in slack channel ?